### PR TITLE
feat: 🎸 paging carousel card

### PR DESCRIPTION
**Pull request by ShadowTourist**
1. Add drag gesture for page control.
2. Add a new carousel detail page.

|video|preview|
|-|-|
|<video src="https://user-images.githubusercontent.com/33440079/129700609-244cefaa-d1f5-4732-85ef-7f97424bc060.mov">|Screen Shot 2021-09-03 at 4 51 03 PM|

### CODE REVIEW

**Review by MarcoEidinger — CHANGES_REQUESTED** [review]
I found one issue that attribute (here: Label/Value4) is not visible on the details page

IssueAttributeNotShown

You can test this by adding the following code to `CAITestApp/CAITestApp/MockData.swift`

```swift
        // text message (user)
        arr.append(CAIResponseMessageData(text: "Show me a single card (technically embedded in carousel", false))

        carouselArr = []
        var cs1 = CAIResponseMessageData("Card Title", "Card SubTitle", "https://miro.medium.com/max/1400/1*j8CdLYOjjiWWWmsXs-48KA.png", "Card Description", iButtons, [UIModelDataSection("Section 1", [UIModelDataValue(value: "Value 1", dataType: "text", rawValue: "Value 1", label: "Label 1", valueState: nil), UIModelDataValue(value: "Value 2", dataType: "text", rawValue: "Value 2", label: "Label 2", valueState: nil), UIModelDataValue(value: "A very long text as value 3 to take call the available space, a very long text as value 3 to take call the available space, a very long text as value 3 to take call the available space indeed.", dataType: "text", rawValue: "A very long text as value 3 to take call the available space", label: "Label 3", valueState: nil), UIModelDataValue(value: "Value 4", dataType: "text", rawValue: "Value 4", label: "Label 4", valueState: nil)])], "Status 1", "Status 2", "Status 3", true)

        carouselArr.append(cs1)
        arr.append(CAIResponseMessageData(carouselArr.map { $0.attachment.content! }, true))
```

Then run CAITestApp and choose `static` Data. You should commit this code as it's good to be able to test the details page without live data :)

Also please fix linting and formatting errors.

**Comment by ShadowTourist** [comment]
@MarcoEidinger  👍 , I will fix it.

File: Sources/SAPCAI/UI/Modifiers/Modifiers.swift [hunk 1]
```diff
@@ -92,3 +93,42 @@ extension View {
...
+                if let (currentValue, action) = state, newValue != currentValue {
+                    action(newValue)
+                }
+                state = (newValue, newAction)
+            }
+    }
+}
+
+public extension View {
+    @_disfavoredOverload
```
Comment (MarcoEidinger) [inline Sources/SAPCAI/UI/Modifiers/Modifiers.swift:126]: why is this needed? doesn't the use of `@_disfavoredOverload` means that the iOS 14  SwiftUI instance method `func onChange<V>(of value: V, perform action: @escaping (V) -> Void) -> some View where V : Equatable` is the preferred one when available? Hence I am wondering if the `if #available` block inside is necessary. I am not sure so just asking...

**Review by MarcoEidinger — CHANGES_REQUESTED** [review]
Please add code to `CAITestApp/CAITestApp/MockData.swift` as earlier proposed.

Also please add description and status to the details page as per the email thread.

File: Sources/SAPCAI/UI/Modifiers/Modifiers.swift (hunk 1 above)
Comment (ShadowTourist) [reply Sources/SAPCAI/UI/Modifiers/Modifiers.swift:126]: I have update drag gesture for carousel page, so I this modifier is unnecessary. I will remove this.

**Comment by ShadowTourist** [comment]
@MarcoEidinger  For description and status, design is not provided yet, so I just add them by the mail showed.

**Comment by ShadowTourist** [comment]
|One Card|Details Status|
|-|-|
|Simulator Screen Shot - iPhone 12 Pro - 2021-09-17 at 15 39 26|Simulator Screen Shot - iPhone 12 Pro - 2021-09-17 at 15 39 29|

**Comment by MarcoEidinger** [comment]
@ShadowTourist design for description and status was changed per email. Please adjust implementation. Thanks!

**Comment by MarcoEidinger** [comment]
According to @ShadowTourist the implementation matches the design so lets get this PR merged, I'll provide build for verification and further feedback has to be uptaken by new PRs

**Review by MarcoEidinger — APPROVED** [review]
